### PR TITLE
Remove dependency of triton during inductor codegen 

### LINKED ISSUE #120192: kernel.codegen_kernel function in inductor without dependency of triton wheel
### 🚀 The feature, motivation and pitch

The kernel. codegen of the inductor is a text generation behavior, and currently it no longer relies on Triton. However, there are still two places where it depends on Triton:
1. gen_attr_descriptor_import : https://github.com/pytorch/pytorch/blob/main/torch/_inductor/codegen/triton.py#L2570
2. signature_of: https://github.com/pytorch/pytorch/blob/main/torch/_inductor/codegen/triton_utils.py#L12 , raw function: https://github.com/openai/triton/blob/main/python/triton/runtime/jit.py#L238

There are currently two reasons to strip Triton's dependencies in the codegen process:
1. [Triton](https://pypi.org/project/triton/#files) currently does not support ARM architecture CPUs in pypi, nor does it have a pure CPU version, [issue](https://github.com/openai/triton/issues/3143). Stripping off related dependencies in the codegen process can allow for certain possibilities in subsequent processes, such as cross compilation.
2. From the perspective of the software itself, stripping the dependency on the actuator in the codegen process is also a reasonable choice.

### Alternatives

_No response_

### Additional context

_No response_

cc @ezyang @msaroufim @bdhirsh @anijain2305 @zou3519

### PR DESCRIPTION
**Pull request by wangjiangben-hw**
Fixes #120192

cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @peterbell10 @ipiszy @yf225 @chenyang78 @kadeng @muchulee8 @aakhundov @ColinPeppler @amjames @jansel

### CODE REVIEW

File: torch/_inductor/utils.py [hunk 1]
```diff
@@ -235,6 +235,38 @@ def next_power_of_2(n: int) -> int:
     return n
 
 
+def _type_of(key):
```
Comment (jansel) [inline torch/_inductor/utils.py:238]: Please add a comment explaining where this function came from.  Is it copied from Triton?  If so add a link.

**Review by jansel — CHANGES_REQUESTED** [review]

File: torch/_inductor/utils.py (hunk 1 above)
Comment (wangjiangben-hw) [reply torch/_inductor/utils.py:238]: I have added a reference link. Please help me take a look. Thank you

**Review by jansel — APPROVED** [review]

**Comment by jansel** [comment]
@wangjiangben-hw please fix the lint errors (`make setup_lint && lintrunner -a`)

**Comment by wangjiangben-hw** [comment]
> @wangjiangben-hw please fix the lint errors (`make setup_lint && lintrunner -a`)

I've passed lintrunner locally, please help me to look at it again~ @jansel

**Comment by jansel** [comment]
I approved the CI run

**Pull request closed without merging**